test/net/sourceforge/kolmafia/session/ChoiceControlTest.java

```diff
@@ -366,6 +366,28 @@ void changingPlaqueRecordsNewName() {
     }
   }
 
+  @Nested
+  class Yachtzee {
+    @Test
+    void visitingTheNCResetsPreference() {
+      var cleanups = new Cleanups(withProperty("encountersUntilYachtzeeChoice ", 0));
+      try (cleanups) {
+        var req = new GenericRequest("adventure.php?snarfblat=404");
+        req.responseText = html("request/test_visit_yachtzee.html");
```

Comment:
You need to actually process the request here using the choice manager.

Tentatively, it looks like the code you added is in `postChoice2`

Reply:
well, in that case you'll need to call `visitChoice` somehow

Reply:
When GenericRequest gets a response from choice.php (whether or not you are visiting the first time or are submitting an option to take the choice, it goes through this sequence of calls in ChoiceManager:

preChoice (before registering the request - which is before submitting the request)
postChoice0 (after getting the response before registering the Encounter)
postChoice1 (before processing results)
(if you are not taking an option, that calls visitChoice)
postChoice2 (after processing results)
(If you are still handling a (chain of) choices, this calls visitChoice)

You don't need to "need to call visitChoice"; that happens automatically.
By all means - set the  property to 20 in visitChoice.
It is pointless to set the property to 19 in BOTH postChoice1 and postChoice2.

It's not obvious why some things have been put in postChoice1 and some in postChoice2.
Some of it is because we want to log something and have it appear before - or after - the logging of items you acquire.
But we update quest progress in either method, remove items from inventory in either, update properties in either, and so on.

Pick one for setting the property to 19.